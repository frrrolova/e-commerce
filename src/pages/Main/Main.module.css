.container {
  position: relative;
}
.container .image-bg {
  position: absolute;
  top: 0;
  z-index: -999;
  width: 100%;
}
.container .section-title {
  margin-top: 3rem;
  text-align: center;
}/*# sourceMappingURL=Main.module.css.map */