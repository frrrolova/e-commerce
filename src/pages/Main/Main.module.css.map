{"version":3,"sources":["Main.module.scss","Main.module.css"],"names":[],"mappings":"AAEA;EACE,kBAAA;ACDF;ADGE;EACE,kBAAA;EACA,MAAA;EACA,aAAA;EACA,WAAA;ACDJ;ADIE;EACE,gBAAA;EACA,kBAAA;ACFJ","file":"Main.module.css"}