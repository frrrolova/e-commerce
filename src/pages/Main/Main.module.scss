@import '../../styles/variables.scss';

.container {
  position: relative;

  .image-bg {
    position: absolute;
    top: 0;
    z-index: -999;
    width: 100%;
  }
}
