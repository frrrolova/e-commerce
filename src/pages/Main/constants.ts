import { Paths } from '@/routes/routeConstants';

export enum PageData {
  TITLE_H1 = 'Bringing Nature Indoors',
  SUBTEXT = 'Step into a lush oasis of tranquility with our exquisite collection of indoor plants, where every leaf tells a story of beauty and natural elegance.',
  TITLE_PROMO = 'Promotions',
  TITLE_DISCOUNT = 'Our special offers',
  TOP_PLANT_ID = '254d8461-038a-4b0a-a9c5-8f1a9343a6c2',
  CATEGORY_ID = '30968be6-31ca-4b72-a1dd-294c99f73c0b',
  TO_CATALOG = 'To Catalog',
}

export enum ButtonLabels {
  LOGIN = 'Login',
  REGISTRATION = 'Registration',
}

export const InfoCardData = {
  heading: 'Summer time',
  imgPath: '/images/home/promo.webp',
  description: 'Take advantage of an additional discount during the summer using the promotional code:',
  subHeading: 'SUMMER2024',
};

export const InfoCardBtn = {
  label: 'To Catalog',
  url: Paths.CATALOG,
};
