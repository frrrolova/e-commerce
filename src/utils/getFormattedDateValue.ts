export function getFormattedDateValue(value: string) {
  const date = new Date(value);
  const dateValue: string = new Date(date.getTime() - date.getTimezoneOffset() * 60000).toISOString().substring(0, 10);
  return dateValue;
}
