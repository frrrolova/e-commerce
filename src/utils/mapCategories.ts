import { Category } from '@/types';
import { Category as CategorySDK } from '@commercetools/platform-sdk';

export function mapCategories(categoryResponse: CategorySDK[]): Category[] {
  return categoryResponse.map((category) => ({
    id: category.id,
    name: category.name['en-US'],
    key: category.key,
    ancestors: category.ancestors,
    parent: category.parent,
  }));
}
