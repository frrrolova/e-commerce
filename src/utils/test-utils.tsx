import React, { PropsWithChildren } from 'react';
import { render } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';

import type { AppStore, RootState } from '../store/store';
// As a basic setup, import your same slice reducers
import initStore from '../store/store';
import { MemoryRouter } from 'react-router-dom';
import { LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { Project } from '@commercetools/platform-sdk';

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: AppStore;
}

const mockShop: Project = {
  countries: [],
  createdAt: new Date().toISOString(),
  languages: [],
  key: 'test-key',
  version: 2,
  name: 'Plant Shop',
  currencies: [],
  messages: { enabled: false, deleteDaysAfterCreation: 15 },
  carts: {
    deleteDaysAfterLastModification: 90,
    countryTaxRateFallbackEnabled: false,
  },
};

export function renderWithProviders(ui: React.ReactElement, extendedRenderOptions: ExtendedRenderOptions = {}) {
  const {
    preloadedState = {
      project: mockShop,
    },
    // Automatically create a store instance if no store was passed in
    store = initStore(preloadedState.project as Project),
    ...renderOptions
  } = extendedRenderOptions;

  const Wrapper = ({ children }: PropsWithChildren) => (
    <MemoryRouter>
      <LocalizationProvider dateAdapter={AdapterDayjs}>
        <Provider store={store}>{children}</Provider>
      </LocalizationProvider>
    </MemoryRouter>
  );

  // Return an object with the store and all of RTL's query functions
  return {
    store,
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
