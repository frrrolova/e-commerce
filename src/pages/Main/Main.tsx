import styles from './Main.module.scss';
import { Container, Typography, Box, Button } from '@mui/material';
import Grid from '@mui/material/Unstable_Grid2';
import ProductCard from '@/components/ProductCard/ProductCard';
import { Product } from '@/types';
import { ButtonLabels, InfoCardData, PageData } from './constants';
import Title from '@/components/Title/Title';
import ImageBg from '/images/home/home-bg.png';
import InfoCard from '@/components/InfoCard/InfoCard';
import { Paths } from '@/routes/routeConstants';
import { useNavigate } from 'react-router-dom';
import { useState, useEffect } from 'react';
import { catalogService } from '@/services/catalogService';

function Main() {
  const navigate = useNavigate();
  const [productTop, setProductTop] = useState<Product | null>(null);
  const [productsOffer, setProductsOffer] = useState<Product[]>([]);
  const [error, setError] = useState<string | null>(null);

  const onRedirect = (url: string) => () => {
    navigate(url);
  };

  useEffect(() => {
    const loadProducts = async () => {
      try {
        const productData = await catalogService.fetchProductById(PageData.TOP_PLANT_ID);
        setProductTop(productData);

        const productsData = await catalogService.fetchProductsByCategory(PageData.CATEGORY_ID);
        setProductsOffer(productsData);
      } catch (err) {
        setError('Failed to fetch products');
        console.log('Failed to fetch products');
      } finally {
        console.log('Data is fetched');
      }
    };

    loadProducts();
  }, []);

  if (error || !productTop) return <Typography color="error">{error}</Typography>;

  return (
    <Box className={styles.container}>
      <Box
        component="img"
        className={styles.imageBg}
        sx={{ display: { xs: 'none', md: 'flex' } }}
        alt="Plant"
        src={ImageBg}
      />

      <Container maxWidth="md" className={styles.containerInner}>
        <Grid container>
          <Grid xs={12} md={8} mt={3}>
            <Typography variant="h1" pt={5} pb={2}>
              {PageData.TITLE_H1}
            </Typography>
            <Typography variant="body1">{PageData.SUBTEXT}</Typography>
            <Button size="small" sx={{ mt: 1, mr: 2 }} onClick={onRedirect(Paths.AUTH)}>
              {ButtonLabels.LOGIN}
            </Button>
            <Button size="small" sx={{ mt: 1 }} onClick={onRedirect(Paths.REGISTER)}>
              {ButtonLabels.REGISTRATION}
            </Button>
          </Grid>
          <Grid xs={12} md={4} mt={3}>
            <ProductCard product={productTop} />
          </Grid>

          {/* PROMO Section */}
          <Grid xs={12} className={styles.sectionTitle}>
            <Title title={PageData.TITLE_PROMO} />
          </Grid>
          <Grid xs={12} mt={3}>
            <InfoCard data={InfoCardData} />
          </Grid>

          {/* DISCOUNT Section */}
          <Grid xs={12} className={styles.sectionTitle}>
            <Title title={PageData.TITLE_DISCOUNT} />
          </Grid>
          <Grid xs={12} mt={3}>
            <Box sx={{ flexGrow: 1 }}>
              <Grid container spacing={{ xs: 2, md: 3 }} columns={{ xs: 4, sm: 8, md: 12 }}>
                {productsOffer.map((product) => (
                  <Grid xs={4} sm={4} md={4} key={product.id}>
                    <ProductCard product={product} />
                  </Grid>
                ))}
              </Grid>
            </Box>
          </Grid>
        </Grid>
      </Container>
    </Box>
  );
}

export default Main;
