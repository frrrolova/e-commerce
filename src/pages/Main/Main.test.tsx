// import { fireEvent, render, waitFor } from '@testing-library/react';
import { renderWithProviders } from '../../utils/test-utils';
import Main from './Main';
// import { ButtonLabels } from './constants';

jest.mock('../../client/client', () => {
  return null;
});

//Mock for useNavigate
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('Main component rendering', () => {
  test('performs snapshot testing', () => {
    const tree = renderWithProviders(<Main />);
    expect(tree).toMatchSnapshot();
  });
});

// describe('Header component behavior', () => {
//   test('should navigate, when clicking on a section title button', async () => {
//     const { getByText } = render(<Main />);

//     await waitFor(() => {
//       const loginButton = getByText(ButtonLabels.LOGIN);

//       fireEvent.click(loginButton);
//       expect(mockNavigate).toHaveBeenCalledTimes(1);
//     });
//   });
// });
