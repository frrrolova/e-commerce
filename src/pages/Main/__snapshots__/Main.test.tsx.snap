// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component rendering performs snapshot testing 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="Plant"
          class="MuiBox-root css-1guk29"
          src="default"
        />
        <div
          class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1pi8xg4-MuiGrid2-root"
          >
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-8 css-4jx1gz-MuiGrid2-root"
            >
              <h1
                class="MuiTypography-root MuiTypography-h1 css-1bvf1x2-MuiTypography-root"
              >
                Bringing Nature Indoors
              </h1>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                Step into a lush oasis of tranquility with our exquisite collection of indoor plants, where every leaf tells a story of beauty and natural elegance.
              </p>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-4 css-1d6tzcd-MuiGrid2-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <img
                      alt="Plant"
                      class="MuiBox-root css-0"
                      src="image.png"
                    />
                    <div
                      class="MuiBox-root css-0"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-ix1vfo-MuiTypography-root"
                      >
                        Product Test1
                      </div>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                      >
                        Description 1
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiBox-root css-0"
                  >
                    <div
                      class="MuiBox-root css-0"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-1coh8cf-MuiTypography-root"
                      >
                        20
                         â‚¬
                      </p>
                    </div>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        aria-label="add"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-16rd89w-MuiButtonBase-root-MuiIconButton-root"
                        data-mui-internal-clone-element="true"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="AddShoppingCartIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2m10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2m-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-16fnxy3-MuiGrid2-root"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              >
                Promotions
              </h4>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-8e9h1c-MuiGrid2-root"
            >
              <div
                class="MuiBox-root css-0"
                data-testid="info"
              >
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
                >
                  <div
                    class="MuiBox-root css-1nnw2ij"
                  >
                    <img
                      alt="Plant"
                      class="MuiBox-root css-0"
                      src="/images/home/promo.webp"
                    />
                    <div
                      class="MuiBox-root css-0"
                    >
                      <div
                        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1r10n4f-MuiTypography-root"
                      >
                        Summer time
                      </div>
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                      >
                        Take advantage of an additional discount during the summer using the promotional code:
                      </p>
                      <h6
                        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                      >
                        SUMMER2024
                      </h6>
                      <div
                        class="MuiBox-root css-0"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-lll1vm-MuiButtonBase-root-MuiButton-root"
                          tabindex="0"
                          type="button"
                        >
                          To Catalog
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-16fnxy3-MuiGrid2-root"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              >
                Our special offers
              </h4>
            </div>
            <div
              class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-8e9h1c-MuiGrid2-root"
            >
              <div
                class="MuiBox-root css-i9gxme"
              >
                <div
                  class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 MuiGrid2-spacing-md-3 css-phooid-MuiGrid2-root"
                >
                  <div
                    class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 MuiGrid2-grid-sm-4 MuiGrid2-grid-md-4 css-rxdc07-MuiGrid2-root"
                  >
                    <div
                      class="MuiBox-root css-0"
                    >
                      <div
                        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
                      >
                        <div
                          class="MuiBox-root css-0"
                        >
                          <img
                            alt="Plant"
                            class="MuiBox-root css-0"
                            src="image2.png"
                          />
                          <div
                            class="MuiBox-root css-0"
                          >
                            <div
                              class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-ix1vfo-MuiTypography-root"
                            >
                              Product Test2
                            </div>
                            <p
                              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                            >
                              Description 2
                            </p>
                          </div>
                        </div>
                        <div
                          class="MuiBox-root css-0"
                        >
                          <div
                            class="MuiBox-root css-0"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 css-1coh8cf-MuiTypography-root"
                            >
                              20
                               â‚¬
                            </p>
                          </div>
                          <div
                            class="MuiBox-root css-0"
                          >
                            <button
                              aria-label="add"
                              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-16rd89w-MuiButtonBase-root-MuiIconButton-root"
                              data-mui-internal-clone-element="true"
                              tabindex="0"
                              type="button"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                                data-testid="AddShoppingCartIcon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2m10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2m-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25"
                                />
                              </svg>
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <img
        alt="Plant"
        class="MuiBox-root css-1guk29"
        src="default"
      />
      <div
        class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
      >
        <div
          class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row css-1pi8xg4-MuiGrid2-root"
        >
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-8 css-4jx1gz-MuiGrid2-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1 css-1bvf1x2-MuiTypography-root"
            >
              Bringing Nature Indoors
            </h1>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              Step into a lush oasis of tranquility with our exquisite collection of indoor plants, where every leaf tells a story of beauty and natural elegance.
            </p>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-4 css-1d6tzcd-MuiGrid2-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
              >
                <div
                  class="MuiBox-root css-0"
                >
                  <img
                    alt="Plant"
                    class="MuiBox-root css-0"
                    src="image.png"
                  />
                  <div
                    class="MuiBox-root css-0"
                  >
                    <div
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-ix1vfo-MuiTypography-root"
                    >
                      Product Test1
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                    >
                      Description 1
                    </p>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-0"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1coh8cf-MuiTypography-root"
                    >
                      20
                       â‚¬
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-0"
                  >
                    <button
                      aria-label="add"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-16rd89w-MuiButtonBase-root-MuiIconButton-root"
                      data-mui-internal-clone-element="true"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="AddShoppingCartIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2m10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2m-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-16fnxy3-MuiGrid2-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            >
              Promotions
            </h4>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-8e9h1c-MuiGrid2-root"
          >
            <div
              class="MuiBox-root css-0"
              data-testid="info"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
              >
                <div
                  class="MuiBox-root css-1nnw2ij"
                >
                  <img
                    alt="Plant"
                    class="MuiBox-root css-0"
                    src="/images/home/promo.webp"
                  />
                  <div
                    class="MuiBox-root css-0"
                  >
                    <div
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1r10n4f-MuiTypography-root"
                    >
                      Summer time
                    </div>
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                    >
                      Take advantage of an additional discount during the summer using the promotional code:
                    </p>
                    <h6
                      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                    >
                      SUMMER2024
                    </h6>
                    <div
                      class="MuiBox-root css-0"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-lll1vm-MuiButtonBase-root-MuiButton-root"
                        tabindex="0"
                        type="button"
                      >
                        To Catalog
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-16fnxy3-MuiGrid2-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            >
              Our special offers
            </h4>
          </div>
          <div
            class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-8e9h1c-MuiGrid2-root"
          >
            <div
              class="MuiBox-root css-i9gxme"
            >
              <div
                class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 MuiGrid2-spacing-md-3 css-phooid-MuiGrid2-root"
              >
                <div
                  class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-4 MuiGrid2-grid-sm-4 MuiGrid2-grid-md-4 css-rxdc07-MuiGrid2-root"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <div
                      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-3su884-MuiPaper-root"
                    >
                      <div
                        class="MuiBox-root css-0"
                      >
                        <img
                          alt="Plant"
                          class="MuiBox-root css-0"
                          src="image2.png"
                        />
                        <div
                          class="MuiBox-root css-0"
                        >
                          <div
                            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-ix1vfo-MuiTypography-root"
                          >
                            Product Test2
                          </div>
                          <p
                            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
                          >
                            Description 2
                          </p>
                        </div>
                      </div>
                      <div
                        class="MuiBox-root css-0"
                      >
                        <div
                          class="MuiBox-root css-0"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1 css-1coh8cf-MuiTypography-root"
                          >
                            20
                             â‚¬
                          </p>
                        </div>
                        <div
                          class="MuiBox-root css-0"
                        >
                          <button
                            aria-label="add"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-16rd89w-MuiButtonBase-root-MuiIconButton-root"
                            data-mui-internal-clone-element="true"
                            tabindex="0"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                              data-testid="AddShoppingCartIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2m10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2m-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25"
                              />
                            </svg>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
